docs/architecture/notifications.md

```
@@ -0,0 +1,558 @@
+---
+
+## Agent Identity
+
+How agents identify themselves when responding on each platform:
+
+| Platform | Per-message identity? | Mechanism |
+|----------|----------------------|-----------|
+| Slack | Yes | `username` parameter in `chat.postMessage` API call |
+| Discord | Yes | `username` parameter in webhook execute |
+| Mattermost | Yes | Bot API username parameter |
+| Telegram | No (fixed bot name) | Agent prefixes message with `[agent-name]: ` |
```

markdownlint-cli2 (0.22.0)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)